### ISSUE TITLE
Setting $debug before first database connection causes fatal exception

### ISSUE THREAD

**Issue by mnewnham** [ADOdb/ADOdb#324]
If $debug mode is set before the first database connection is successfully added to the load balancer group it causes the the load balancer to throw a fatal exception.  This means that debugging mode cannot be used to identify connection problems
Valid code:
````
$driver = 'mysqli'
 $db = new ADOdbLoadBalancer( $driver );

$database = 'employees';
$user     = 'adodb';
$password = 'adodb';
$host     = '192.168.0.200';
$db_connection_obj = new ADOdbLoadBalancerConnection( $driver,'write', 10, false, $host, $user, $password, $database ); //Master with weight of 10
$db_connection_obj->getADODbObject(); //Pass specific settings to the ADOdb object itself.
$db->addConnection( $db_connection_obj );

$db->debug = true;
````
Invalid Code:
````
$driver = 'mysqli'
$db = new ADOdbLoadBalancer( $driver );
$db->debug = true;

$database = 'employees';
$user     = 'adodb';
$password = 'adodb';
$host     = '192.168.0.200';
$db_connection_obj = new ADOdbLoadBalancerConnection( $driver,'write', 10, false, $host, $user, $password, $database ); //Master with weight of 10
$db_connection_obj->getADODbObject(); //Pass specific settings to the ADOdb object itself.
$db->addConnection( $db_connection_obj );
````
Result
````
Fatal error: Uncaught exception 'Exception' with message 'Unable to return Connection object...' in C:\dev\github\lb\adodb-loadbalancer.inc.php on line 252

Exception: Unable to return Connection object... in C:\dev\github\lb\adodb-loadbalancer.inc.php on line 252

Call Stack:
    0.0003     360360   1. {main}() C:\dev\github\tests\loadbaltest.php:0
    0.0085    2162336   2. ADOdbLoadBalancer->__set() C:\dev\github\tests\loadbaltest.php:31
    0.0085    2162336   3. ADOdbLoadBalancer->getConnection() C:\dev\github\lb\adodb-loadbalancer.inc.php:571

````

**Comment by ipso** [comment]
The ADOdbLoadBalancer class doesn't directly handle debug, since its essentially just a router that sits above ADODB objects and simply passes calls through to the real ADODB objects as necessary. That is why you see the uncaught exception, since its trying to pass $db->debug = true; through to some ADODB object, but one hasn't been defined yet.

So if you wanted to debug connection problems you would do something like this:

```
$driver = 'mysqli'
$db = new ADOdbLoadBalancer( $driver );

$database = 'employees';
$user     = 'adodb';
$password = 'adodb';
$host     = '192.168.0.200';
$db_connection_obj = new ADOdbLoadBalancerConnection( $driver,'write', 10, false, $host, $user, $password, $database ); //Master with weight of 10

$db_connection_obj->getADODbObject()->debug = true; //Enable debug mode for this connection 
$db->addConnection( $db_connection_obj );
```

This way you can also control debugging to specific connections instead of them all if needed. Though as you discovered once a connection has been established, you can call $db->debug = TRUE; too.

We could teach the load balancer about $debug before any connections are made, but I believe it would add a lot of additional complexity and a performance hit when the same behavior can be easily achieved by simply reordering the commands. Additionally it would also cause an opposite effect of how do handle settings defined at the load balancer level vs. the ADODB object level? Which takes precedence and across what connections? How would they be synchronized and handled sanely?

**Comment by mnewnham** [comment]
OK, I understand that, and I've clearly made a mistake that I suspect lots of people are going to make, and for exactly the same reason that I did it, which was to try to work out why the load balancer wasn't executing a command even though no errors had been reported. The reason was, of course, that the password on the readonly server was incorrect, and the first thing I'd do would be to enable debugging at that the point I did. Is there perhaps a way that we can throw a warning if $db->debug is set prior to connection, instead of it excepting.
